tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -13634,12 +13682,15 @@ SpirvEmitter::loadDataFromRawAddress(SpirvInstruction *addressInUInt64,
   assert(loadInst);
   loadInst->setAlignment(alignment);
   loadInst->setRValue();
+  loadInst->setLayoutRule(SpirvLayoutRule::Physical);
```

Comment:
Should `address` get the layout rule set as well? I wonder if this will remove the need for the change in `lowerType`, because it will now get the correct layout rule.

Reply:
Yes, thought changing the incoming instruction was weird, but it is done for the storage class, so makes sense.
And yes, it allows me to remove the weird thing in lowerTypeVisitor.

Reply:
This does not change the storage class for the incoming instruction. It sets it for the BitCast that was just created to make sure you are casting to the correct type.